eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala

```diff
@@ -162,7 +162,15 @@ class PeerConnection(keyPair: KeyPair, conf: PeerConnection.Conf, switchboard: A
     heartbeat {
       case Event(msg: LightningMessage, d: ConnectedData) if sender() != d.transport => // if the message doesn't originate from the transport, it is an outgoing message
         d.transport forward msg
-        stay()
+        msg match {
+          case _: OnionMessage if !d.isPersistent =>
+            startSingleTimer(KILL_IDLE_TIMER, KillIdle, conf.killIdleDelay)
+            stay()
+          // If we send any channel management message to this peer, the connection should be persistent.
+          case _: ChannelMessage if !d.isPersistent =>
```

Comment:
As I commented in the other PR, treating all channel messages equally is too easy to game: https://github.com/ACINQ/eclair/pull/2092#discussion_r766664266

Reply:
But as a first step for the non-malicious cases, this is a good improvement, so we can address that later.